DiscordBot/Services/CommandHandlingService.cs

```diff
@@ -143,17 +144,44 @@ private bool TryDequeue(string prefix, out CommandInfo info)
         private async Task OnMessageReceived(SocketMessage arg)
         {
             var result = ParseCommand(arg as SocketUserMessage);
-            
+
             if (result.HasValue)
             {
                 if (result.Value.SearchResult.IsSuccess)
                 {
-                    await result.Value.Context.Message.AddReactionAsync(BotReaction.InProgress);
+                    Boolean isLimited = false;
+                    List<long> toRemove = new List<long>();
 
-                    if (!TryEnqueue(result.Value))
+                    foreach (KeyValuePair<long, ulong> entry in RecentUsers)
+                    {
+                        if(entry.Key < DateTime.Now.Ticks - 600000000)
+                        {
+                            toRemove.Add(entry.Key);
+                        }
+                        else if (entry.Value == arg.Author.Id)
+                        {
+                            isLimited = true;
+                        }
+                    }
+                    foreach(long key in toRemove)
                     {
-                        await result.Value.Context.Message.RemoveReactionAsync(BotReaction.InProgress, Discord.CurrentUser);
-                        await result.Value.Context.Message.AddReactionAsync(BotReaction.Error);
+                        RecentUsers.Remove(key);
+                    }
+
+                    if (isLimited)
+                    {
+                        await result.Value.Context.Message.AddReactionAsync(BotReaction.RateLimited);
+                    }
+                    else
+                    {
+                        RecentUsers.Add(DateTime.Now.Ticks, arg.Author.Id);
```

Comment:
If there is the same time, an ArgumentException will be thrown.
Because the OnMessageReceived method is async, it can be executed at the same time.